File: app/src/main/java/com/pietrantuono/podcasts/addpodcast/view/ApiLevelCheckerImpl.java
package com.pietrantuono.podcasts.addpodcast.view;

import android.content.Context;
import android.os.Build;

public class ApiLevelCheckerImpl implements ApiLevelChecker {

    @Override
    public boolean isLollipopOrHigher() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
    }
}

File: app/src/main/java/com/pietrantuono/podcasts/addpodcast/view/ApiLevelCheckerImpl.kt
package com.pietrantuono.podcasts.addpodcast.view

import android.os.Build

class ApiLevelCheckerImpl : ApiLevelChecker {

    override val isLollipopOrHigher: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP

    override val isMarshmallowOrHigher: Boolean
        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
}
